errbot/core.py

```diff
@@ -158,7 +158,14 @@ def send_templated(self, identifier, template_name, template_parameters, in_repl
         return self.send(identifier, text, in_reply_to, groupchat_nick_reply)
 
     def split_and_send_message(self, msg):
-        for part in split_string_after(msg.body, self.bot_config.MESSAGE_SIZE_LIMIT):
+        size_limit = self.bot_config.MESSAGE_SIZE_LIMIT
+
+        # If MESSAGE_SIZE_LIMIT_INCLUDES_ADDRESEE is set to True, lower the
+        # size_limit by the size of the string representation of the addressee
+        if self.bot_config.MESSAGE_SIZE_LIMIT_INCLUDES_ADDRESEE:
+            size_limit -= len(str(msg.to))
```

Comment:
How about taking a slightly different approach of modifying the signature of `split_and_send_message` to:

```python
def split_and_send_message(self, msg, max_length=None):
    if max_length is None:
        max_length = self.bot_config.MESSAGE_SIZE_LIMIT

    for part in split_string_after(msg.body, size_limit):
        partial_message = msg.clone()
        partial_message.body = part
        self.send_message(partial_message)
```

You can then do the `max_length -= len(str(msg.to))` calculation in the IRC backend directly and pass `max_length` to `split_and_send_message` explicitly. That way it's completely self-contained within the IRC backend.

Reply:
@zoni Sorry for my late reply here.

Your solution would indeed be much better but sadly, `split_and_send_message` is not called from the IRC backend. The IRC backend could still set some sort of object variable/flag on `IRCBackend` (which inherits from `ErrBot` which in turn contains `split_and_send_message`) that would cause the `split_and_send_message` function to execute code similar to what there currently is in the PR, without the global config option.

I may also not fully understand what you meant, so please feel free to point out if I missed anything.

Thanks!

Reply:
@mrshu Hey! Following up for @zoni 

I think the goal here is trying to keep this specific IRC code in the backend. While `split_and_send_message` is not called from IRCBackend, I agree that it still could be self-contained in the backend.

I recommend looking at the SlackBackend's implementation for their size limits:

https://github.com/errbotio/errbot/blob/fe64c2514cdd274b7ee14c80ea47598f88db70e8/errbot/backends/slack.py#L651-L668

it's completely self contained. Is there a reason to have a toggleable option for additional size limitation? Wouldn't all IRC messages contain ADDRESEE? or is that only for PRIVMSG? It's bene a while since I've interacted with the IRC backend.